src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharsBitmap.cs

```diff
@@ -21,9 +21,9 @@ private AllowedCharsBitmap(uint[] allowedCharsBitmap)
         // Marks a character as allowed (can be returned unencoded)
         public void AllowCharacter(char c)
         {
-            uint codePoint = (uint)c;
-            int index = (int)(codePoint >> 5);
-            int offset = (int)(codePoint & 0x1FU);
+            int codePoint = c;
+            int index = codePoint >> 5;
```

Comment:
Is this computation safe with two's complement? My two's complement knowledge is super rusty (I learned that what, like, 15 years ago?? :smile: ). That is, for "very large" numbers that are positive in `uint`, but negative as `int`, will `>>` maintain the sign after the shift? E.g. does `1000 0000 0000 0000 >> 5` become `1111 1100 0000 0000` or `0000 0100 0000 0000`?